aiida/cmdline/commands/cmd_computer.py

```diff
@@ -161,6 +162,55 @@ def _computer_create_temp_file(transport, scheduler, authinfo):  # pylint: disab
     return True, None
 
 
+def time_use_login_shell(authinfo, auth_params, use_login_shell: bool) -> float:
+    """Execute the ``whoami`` over the transport for the given ``use_login_shell`` and report the time taken."""
+    import time
+
+    auth_params['use_login_shell'] = use_login_shell
+    authinfo.set_auth_params(auth_params)
+
+    time_start = time.time()
+
+    with authinfo.get_transport() as transport:
+        transport.whoami()
+
+    return time.time() - time_start
+
+
+def _computer_use_login_shell_performance(transport, scheduler, authinfo):  # pylint: disable=unused-argument
+    """Execute a command over the transport with and without the ``use_login_shell`` option enabled.
+
+    By default, a computer is configured to use a login shell when connecting to it using the transport. For certain
+    environments, loading the login script of the shell is quite costly which will significantly slow down all commands
+    executed. This has a big impact on the throughput of calculation jobs. This test tries to execute a simple command
+    both with and without using a login shell and emits a warning if the ratio of the execution time exceeds 2.
```

Comment:
```suggestion
    By default, AiiDA uses a login shell when connecting to a computer in order to operate in the same environment as a user connecting to the computer.
    However, loading the login scripts of the shell can take time, which can significantly slow down all commands
    executed by AiiDA and impact the throughput of calculation jobs. 
    This test executes a simple command both with and without using a login shell and emits a warning if the ratio of the execution times exceeds 2.
    If the computer is already configured to avoid using a login shell, the test is skipped and the function returns a successful test result.
```